google/cloud/spanner_dbapi/connection.py

```diff
@@ -259,14 +314,18 @@ def commit(self):
 
         This method is non-operational in autocommit mode.
         """
+        self._snapshot = None
+
```

Comment:
I think it would make more sense to add the `read_only` check here and exit early. If the connection is read only, then none of the code below should be relevant. WDYT?

This also applies to `rollback()`.

Reply:
The `self._release_session()`call would also be needed in the early exit branch in that case.